fixtures/tests/_fixtures/test_monkeypatch.py

```diff
@@ -20,9 +20,20 @@
 reference = 23
 
 class C(object):
+    def foo(self): pass
     @staticmethod
-    def foo(): pass
-def bar(): pass
+    def foo_staticmethod(): pass
+    @classmethod
+    def foo_classmethod(cls): pass
+
+def fake(arg): pass
+def fake2(arg): pass
+def fake_no_args(): pass
+@staticmethod
```

Comment:
My opinion is that if F.foo is patched onto another class I would expect the assert to fail. If cls is already bound to the class where foo is defined that makes it more difficult to define a classmethod to patch on to F where properties of F should be accessible.


As you may have noticed in the tests I only checked the direction of taking functions and patching them onto a class. I did not consider taking a staticmethod of a class and patching a regular function. I completely agree that it would be odd for someone to do that and hopefully it would raise an eyebrow in a review. I don't think that behavior has changed with this patch, or the previous patch, because most of the logic is behind a check that the patch target is defined on a class.

I'm happy to add some tests going in the other direction if there are conclusions on what the behavior should be. At this point I'm leaning towards patching fake_no_args with C.foo_staticmethod being crazy, but not something that should be explicitly disallowed.